From 96dbbd4345810da3b4a62d3fdbdfcc0e7d848838 Mon Sep 17 00:00:00 2001
From: Kulwinder Singh <kulwinder.singh@mediatek.com>
Date: Fri, 31 Aug 2018 13:14:03 +0530
Subject: [PATCH 37/45] Fix for CTS testAllowWhileIdleJobInTempwhitelist

In this test case when all constraints to run the jobs are satisfied,
then to make job ready to run, intent MSG_CHECK_JOB is send to job handler.
But this intent is not dispatched to job handler.

BUG: 113778907
Test:After these changes this CTS case always passed.

Change-Id: I8c63361dcd7ade19b724c9bc2fdf78d5f10c8d71
Merged-In: I8c63361dcd7ade19b724c9bc2fdf78d5f10c8d71

Feature: CTS(*)
(cherry picked from commit de0a0155762d381c435085d11245f501ad59921b)

Change-Id: Ie8dc8518910a9a87eb88ace1e6dac022d9faa909
---
 .../core/java/com/android/server/job/JobSchedulerService.java   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/job/JobSchedulerService.java b/services/core/java/com/android/server/job/JobSchedulerService.java
index 7e23ee9b4be..6608cad19d6 100644
--- a/services/core/java/com/android/server/job/JobSchedulerService.java
+++ b/services/core/java/com/android/server/job/JobSchedulerService.java
@@ -1610,6 +1610,7 @@ public class JobSchedulerService extends com.android.server.SystemService
                         }
                     } break;
                     case MSG_CHECK_JOB:
+                        removeMessages(MSG_CHECK_JOB);
                         if (mReportedActive) {
                             // if jobs are currently being run, queue all ready jobs for execution.
                             queueReadyJobsForExecutionLocked();
@@ -1666,7 +1667,6 @@ public class JobSchedulerService extends com.android.server.SystemService
                 }
                 maybeRunPendingJobsLocked();
                 // Don't remove JOB_EXPIRED in case one came along while processing the queue.
-                removeMessages(MSG_CHECK_JOB);
             }
         }
     }
-- 
2.17.1

